test_conformance/computeinfo/extended_versioning.cpp

```diff
@@ -688,14 +698,41 @@ static int test_extended_versioning_device_built_in_kernels(cl_device_id device)
     return 0;
 }
 
+// Assumes the core enums, structures, and macros exactly match
+// the extension enums, structures, and macros:
+
+static_assert(CL_PLATFORM_NUMERIC_VERSION == CL_PLATFORM_NUMERIC_VERSION_KHR,
+              "CL_PLATFORM_NUMERIC_VERSION mismatch");
+static_assert(CL_PLATFORM_EXTENSIONS_WITH_VERSION
+                  == CL_PLATFORM_EXTENSIONS_WITH_VERSION_KHR,
+              "CL_PLATFORM_EXTENSIONS_WITH_VERSION mismatch");
+
+static_assert(CL_DEVICE_NUMERIC_VERSION == CL_DEVICE_NUMERIC_VERSION_KHR,
+              "CL_DEVICE_NUMERIC_VERSION mismatch");
+static_assert(CL_DEVICE_EXTENSIONS_WITH_VERSION
+                  == CL_DEVICE_EXTENSIONS_WITH_VERSION_KHR,
+              "CL_DEVICE_EXTENSIONS_WITH_VERSION mismatch");
+static_assert(CL_DEVICE_ILS_WITH_VERSION == CL_DEVICE_ILS_WITH_VERSION_KHR,
+              "CL_DEVICE_ILS_WITH_VERSION mismatch");
+static_assert(CL_DEVICE_BUILT_IN_KERNELS_WITH_VERSION
+                  == CL_DEVICE_BUILT_IN_KERNELS_WITH_VERSION_KHR,
+              "CL_DEVICE_BUILT_IN_KERNELS_WITH_VERSION mismatch");
+
+static_assert(sizeof(cl_name_version) == sizeof(cl_name_version_khr),
```

Comment:
Mabye you could use `std::is_same` or something a bit stronger than just comparing sizes.

Reply:
We can't use `std::is_same` since they aren't the same type, though they do have the same structure fields in the same order.

Reply:
Ouch, I thought we used typedefs. We probably ought to fix this in the headers.